Src/java/cql/src/main/java/org/cqframework/cql/Application.kt

```diff
@@ -0,0 +1,25 @@
+package org.cqframework.cql
+
+import java.io.FileInputStream
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.CommonTokenStream
+import org.cqframework.cql.gen.cqlLexer
+import org.cqframework.cql.gen.cqlParser
+
+object Application {
+    fun main(args: Array<String>) {
+        val inputFile = args.getOrNull(0)
+        val stream = inputFile?.let { FileInputStream(it) } ?: System.`in`
+
+        val chars = CharStreams.fromStream(stream)
+        val lexer = cqlLexer(chars)
+        val tokens = CommonTokenStream(lexer)
+        val parser = cqlParser(tokens)
+
+        parser.buildParseTree = true
+        val tree = parser.library()
+
+        // show tree in text form
+        println(tree.toStringTree(parser))
+    }
```

Comment:
Since this is not a static function, if you try to run it via `gradle :cql:run`, it will fail with an error like:
> Error: Main method is not static in class org.cqframework.cql.Application, please define the main method as:
>     public static void main(String[] args)

Based on some googling, I found two approaches that work.

One approach is to declare `Application` as a class with a companion object containing an annotated main function. E.g.:
```kt
class Application {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            // all the good stuff goes here
        }
    }
}
```

The other option is to drop the wrapping `Application` object entirely and just define `main` as a standalone function (not inside an object or class). In this case, however, you'll need to change the `mainClass` in `cql/build.gradle.kts` like this:
```gradle
application {
    mainClass = "org.cqframework.cql.ApplicationKt"
}
```

I tried both approaches locally via `gradle :cql:run --args="/path/to/my/cql/file"` and they seem to work. I don't know Kotlin at all, so maybe one way is preferable to the other. I'll leave that to you to decide.

Reply:
I'll do your first alternative for the sake of providing a reference in the code base for those new to Kotlin on companion objects and the @JvmStatic annotation.

Reply:
Updated.